erts/etc/win32/wsl_tools/vc/emu_cc.sh

```diff
@@ -23,6 +23,10 @@
 
 if [ X"$CONFIG_SUBTYPE" = X"win64" ]; then
     GCC="x86_64-w64-mingw32-gcc -m64"
+elif [ X"$CONFIG_SUBTYPE" = X"arm64" ]; then
+    GCC="aarch64-w64-mingw32-gcc -m64"
+elif [ X"$CONFIG_SUBTYPE" = X"x64_arm64" ]; then
+    GCC="aarch64-w64-mingw32-clang -m64"
```

Comment:
Hi @jhogberg, `x64_arm64` is used when cross-compiling to arm64 windows on an x86_64 machine because we need to call `vcvarsall.bat` in `erts/etc/win32/wsl_tools/SetupWSLcross.bat` with different parameters to get right msvc toolchain, for example,

- call `vcvarsall.bat arm64` if compile natively
- call `vcvarsall.bat amd64_arm64` if cross-compile on x86_64 machines

And I think that it's somewhat awkward naming on Windows because `win32` and `win64` traditionally only stands for 32-bit and 64-bit respectively, but they actually means `x86` and `x86_64`. I wonder if we should standardise these naming while providing "aliases" so that build script won't break. I'm think that adding a universal entry point for setting up environment variables:

`./otp_build env_win [<BUILD_ARCH>-]<TARGET_ARCH>`

where `BUILD_ARCH` and `TARGET_ARCH` each can be one of `x86`, `x86_64` and `aarch64`.

For example,

### x86 target
For compiling natively on x86 windows, `env_win32` should be the alias of `env_win x86` (or vice-versa). The following commands should behave the same

- `./otp_build env_win x86`, universal entry point
- `./otp_build env_win32`, alias for backward compatibility

For cross-compiling, we will have 
- `./otp_build env_win x86_64-x86`, universal entry point
- `./otp_build env_win aarch64-x86`, universal entry point

### x86_64 target
Similarly, the following commands should behave the same

- `./otp_build env_win x86_64`, universal entry point
- `./otp_build env_win32 x64`, alias for backward compatibility
- `./otp_build env_win32 x64`, alias for backward compatibility
- `./otp_build env_win64 x64`, alias for backward compatibility

For cross-compiling, we will have 
- `./otp_build env_win x86-x86_64`, universal entry point
- `./otp_build env_win aarch64-x86_64`, universal entry point

### aarch64 target
For native compiling,
- `./otp_build env_win aarch64`, universal entry point

Maybe we can have these aliases which would follow the convention:
- `./otp_build env_win32 x86-aarch64`
- `./otp_build env_win32 x64-aarch64`
- `./otp_build env_win64 x86-aarch64`
- `./otp_build env_win64 x64-aarch64`

And for cross-compiling, we will have 
- `./otp_build env_win x86-aarch64 `, universal entry point
- `./otp_build env_win x86_64-aarch64`, universal entry point

WDYT?

Reply:
The reason why using `clang` for cross-compiling is that `aarch64-w64-mingw32-gcc` seems to crash the windows machine😅 I'll double verify this on another x86_64 windows machine.

Reply:
> What GCC version is `aarch64-w64-mingw32-gcc`? There's a known bug in GCC 7 (maybe 8 or 9 too?) where it segfaults while compiling the JIT. If so, I don't think we want this workaround.

It's actually LLVM with a wrapper script, https://github.com/mstorsjo/llvm-mingw/releases. I tested `aarch64-w64-mingw32-gcc` on a physical x86_64 machine, and it seems to be okay now (as I'm not able to do a full build -- there're still some issues when cross-compiling). 

I think the reason was that I was testing on a virtual machine, and perhaps there were some bugs in the virtualization layer/software.